Formula/inspectrum.rb

```diff
@@ -20,9 +20,17 @@ class Inspectrum < Formula
   depends_on "liquid-dsp"
   depends_on "qt@5"
 
+  on_linux do
+    depends_on "gcc"
+  end
+
+  fails_with gcc: "5"
```

Comment:
for my own knowledge, what does `fails_with` convey that `depends_on "gcc"` doesn't?

Reply:
I think it's an indication that Linux can't use the system GCC if it's GCC 5.

Reply:
gotcha, but doesn't `depends_on "gcc"` mean we know exactly what version we will use and therefore no longer care about the system's `gcc`?

for comparison, when we `depends_on "python@3.9"` we don't need to say `fails_with python: "2"` etc. granted i'm not very familiar with c-based languages so i am probably just missing something small here

Reply:
That is my understanding too. For our bottling:
- On Linux, we build on Ubuntu 16.04 which has `gcc-5`. In future, we will move to Ubuntu 18.04 which has `gcc-7`.
- On macOS, we build with Apple `clang`.

Unless you add `fails_with`, Homebrew Linux CI will pick the defaults during compiler selection.

Once you add `fails_with`, Homebrew will exclude it, which means another compiler needs to be added (e.g. `depend_on "gcc"`) or build will fail.

When Linux CI is moved to Ubuntu 18.04, we will be able to drop `gcc` dependency on a lot of formulae.

---

The situation of adding `gcc` without `fails_with` is usually for something else, like `gfortran`.

Reply:
ah i see now, thanks for explaining!